### DISCUSSION TITLE
Default Values on custom fields

### DISCUSSION THREAD

**Discussion by damo2929** [netbox-community/netbox#12808] [Help Wanted!]
I have a custom field linked to VLAN and I want to have a default value of NET1
the field is requiring json but I have no Idea whats it's wanting from me

I have tried 
```
{
"vid": 2
}
```

but get 
```
Invalid default value "{'vid': 2}": Value must be an object ID, not dict
```

how would I set the this ?

**Comment by candlerb** [comment]
This is a custom field of type "Object", correct?

> Value must be an object ID

That means the default value you enter must be the raw object ID, e.g. the number 2.

`2` is a valid JSON value, so that's what you put directly in the field. (Without any quotes: the JSON encoding of the number 2 is just the digit `2` by itself)

Do remember though, that the VLAN with ID 2 is not the same as vid 2.  vid is the 802.1q tag, from 1-4094.  The Netbox ID is the database ID, so specifying `2` here means whatever VLAN you see at `/ipam/vlans/2/` in the GUI.  If this is your "NET1" then that's what you want (of course, it may have a vid other than 2)

Also, you can have multiple VLANs in your database all with the same vid. This is quite common, e.g. where you have different sites with their own distinct and separate VLANs but using the same tag.

**Comment by dejantep** [comment]
I'm having difficulties with this.

I use custom field type "Text" and woud like to display some default text value for the user but when rendered it shows as JSON.
For instance:

```
{
    "BNID": "Leave empty for now"
}
```

shows this in gui {'BNID': 'Leave empty for now'}
